### PR TITLE
Milestone 2: To Do list: interactive list

### PR DESCRIPTION
**Pull request by den-cloud**
Added 2 new JavaScript files and imported them as modules:
one contains methods related to drag/drop and sorting functionality.
the other will contain methods related to the status updates (completed: true/false).
Added event listeners to list items (dragstart, dragover, drop) and to checkbox (change).
Updated items object's values for index and completed keys upon user actions.
Stored the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: src/services/update.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const sortTodo = (todos) => todos.sort((a, b) => a.index - b.index);
+export const save = (todos) => localStorage.setItem('todos', JSON.stringify(todos));
+
+export const saveToStorage = (todos) => {
+  const sortedTodos = sortTodo(todos);
+  save(sortedTodos);
+};
```
Comment (dasileker) [inline src/services/update.js]: - [ ] Your method for storing the array of object is good, but you still more to make it work.

jdjdjd

File: src/services/update.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  return todos;
+};
+
+export const getTodos = () => {
+  const todos = localStorage.getItem('todos');
+  if (todos) {
+    return JSON.parse(todos);
+  }
+  return [];
+};
```
Comment (dasileker) [inline src/services/update.js]: - [ ] Your method for storing the array of object and sorting it, is good but you still more to make it work.

jjdj

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @den-cloud,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by wandji20 — APPROVED** [review]
## Hi @den-cloud :raised_hands:

##### You have done exceptionally well, and have met all requirements for this project:+1:
:100:  Great job creating re-usable functions
# Status: Approved 👍 :partying_face: 

Your project has been approved, however, Keep up the good work.

## Congratulations :+1:  :fireworks: 

_Go ahead and merge this branch. Also, don't forget to submit the project completion form.
Good luck with your next project :mechanical_arm:  :slightly_smiling_face:_ 

 If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**